datafusion/physical-expr/src/expressions/cast.rs

```diff
@@ -201,7 +214,10 @@ pub fn cast_with_options(
     let expr_type = expr.data_type(input_schema)?;
     if expr_type == cast_type {
         Ok(expr.clone())
-    } else if can_cast_types(&expr_type, &cast_type) {
+    } else if can_cast_types(&expr_type, &cast_type)
```

Comment:
I think we need to have a followup PR in arrow-rs, I'll do it

Reply:
Yes, if this can be pushed down to the arrow crate, the complexity in datafusion would be reduced. I wasn't sure if doing so was appropriate

Reply:
I just checked arrow-rs so such cast is supported https://github.com/apache/arrow-rs/blob/master/arrow-cast/src/cast.rs#L224

Reply:
`can_cast_types` was returning false in my testing yesterday for `Float64` -> `Timestamp(Nanosecond, None)`, seemingly because the line you linked has not been released yet. The type check was changed from `is_integer` to `is_numeric` [only a few days ago](https://github.com/apache/arrow-rs/commit/435b53ded3710216e32ad3f82ae4910f50954e06#diff-50ca33c0aee08395003926d3f7c2d7ee6d90199dea0f2642856cfac201e27a68L226-R225), whereas the last arrow-rs release was 3 weeks ago.

Should we wait until the next arrow-rs release so I can leverage this change?

Reply:
I'm okay to let it go, because this is important piece. I'll create a followup issue to move to arrow-rs cast and small other refactoring. Thanks @spaydar for your work